zip-tor-txn-relay-network.rst

```diff
@@ -0,0 +1,132 @@
+    - Where the same `Sapling Address` appears multiple times, remove all but the most recent entry.
+    - Where the same `Tor Hidden Service` appears multiple times, remove all but the most recent entry.
+- When a Wallet intends to send a shielded transfer, it:
+    - Selects a uniformly random entry in the `deduplicated list`. Wallets must use uniform random selection and not rely on local information such as how reliable a provider is.
+    - Includes a Sapling Shielded Output to the given Sapling Address transferring at least `Relay Fee` ZEC. The memo field must be empty and is reserved.
+    - Generates necessary proofs and signs the complete transaction.
+    - Connects to the hidden service and executes the `TTRN Relay Protocol` as a client.
+- When a provider receives a transaction via the Hidden Service TTRN Relay Protocol service, it decodes the transfer to its Sapling Address.
+    - If the fee paid is at least `Relay Fee`, it submits the transaction to the Zcash network vai the P2P protocol.
+    - If the fee paid is less, it drops the transaction.
+
+Security Considerations
+=======================
+
+TODO
+
+- unfinished thoughts:
+    - what is different in privacy protections for wallets between there being 1 provider versus k providers who do not share information? Or maybe a better framing: p providers are malicious and 1-p are not.
+    - malicious providers can passively surveil, but what else can they do? Drop selected transactions?
+    - All of my thinking so far has assumed fully shielded transactions. What about partially shielded transactions?
+
+Economic Considerations
+=======================
+
+TODO: refine/harden/analyze these:
+
+- We call the total number of transactions routed through TTRN in a given time period times the relay fee is `aggregate revenue`.
+- Since active providers are chosen uniformly randomly, the `expected revenue per provider` (aka `ERPP`) in the time period is `aggregate revenue` / `number of registrations`.
+- Over time, given stable conditions like a stable transaction rate, we expect the number of providers will increase or decrease such that `ERPP` + `non-financial incentive` + `out-of-band financial incentives` = `Total Cost for Service` + `epsilon` for providers.
+    -  We include `non-financial incentive` to capture the notion that some providers may have other incentives to overcome excessive costs, such as the altruism of improving privacy for users or to execute an attack on privacy. However, we posit that over long enough time frames the non-financial incentives will tend towards 0 or else the providers become bankrupt and cannot continue.
+    - We include `out-of-band financial incentives` to capture cases where providers earn other revenue in connection to their service that isn't in the scope of this protocol. For example, as long as a charity pays providers for their service separately from txn fees, this can tip the balance financially for those providers.
```

Comment:
I don't know how.

Actually here's a wacky thought: force all providers to publish all local/privileged information globally: publish their viewkey (and maybe even a log of Tor metadata). Now they can't sell that information, and it's no longer "privileged" metadata. I'm genuinely curious what kind of threat model this may support, because it would seem on the face of it to remove any incentive to attack TTRN providers except for active attacks like dropping/malleating txns. Maybe in this model the security is the same for 1 vs N providers.